src/Microsoft.Azure.Batch.SoftwareEntitlement.Common/CertificateStore.cs

```diff
@@ -84,13 +84,23 @@ public Errorable<X509Certificate2> FindByThumbprint(string purpose, CertificateT
                 from location in _storeLocations
                 select FindByThumbprint(thumbprint, name, location);
 
-            var certificate = query.FirstOrDefault(cert => cert != null);
-            if (certificate == null)
+            var candidates = query.Where(cert => cert != null).ToList();
+
+            var certWithPrivateKey = candidates.Find(cert => cert.HasPrivateKey);
+            if (certWithPrivateKey != null)
+            {
+                // We might have multiple copies of the certificate available
+                // If so, prefer any copies that have their private key over those that do not
```

Comment:
Wonder if it's worth including "why" in the comment here i.e. why are certs with their private key preferred and is it an issue if the selected cert doesn't include its private key?